src/parser/autolink.rs

```diff
@@ -373,6 +375,13 @@ fn autolink_delim(data: &str, mut link_end: usize, relaxed_autolinks: bool) -> u
             }
 
             link_end -= 1;
+        } else if cclose == LINK_END_UNICODE[2] {
+            let slice = &bytes[link_end - LINK_END_UNICODE.len() .. link_end];
+
+            if slice == LINK_END_UNICODE {
+                link_end -= LINK_END_UNICODE.len();
+                break;
+            }
```

Comment:
```suggestion
            }
            break;
```

Without the `break;` (per the catch-all `else`), any encounter of a trailing 0xa9 byte that isn't part of U+2069 POP DIRECTIONAL ISOLATE will cause an infinite loop. e.g.:

```console
$ printf 'www.copyrighted.com\xc2\xa9' | cargo run -- -e autolink
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/comrak -e autolink`
```